Functions/QualityControl/xASL_qc_CollectParameters.m

```diff
@@ -34,37 +35,43 @@
     error('Missing iSubject...');
 end
 
+% It is recommended to provide session number for ASL
+if nargin<4 || isempty(iSession)
+	iSession = 1;
+	if strcmp(ScanType, 'ASL')
+		warning('Missing iSession for ASL, setting to 1');
+	end
+end
+
 if ~isfield(x,'Output')
     x.Output  = struct; 
 end
-if ~isfield(x.Output,ScanType)
+if ~isfield(x.Output, ScanType)
     x.Output.(ScanType) = struct; 
 end
 
+%% -----------------------------------------------------------------------------------------------
+%% Collect subject-specific (i.e. structural/anatomical) QC results
+
 switch ScanType
     case 'Structural'
-        QCCollectFunction = @xASL_qc_CollectQC_Structural;
+        x = xASL_qc_CollectQC_Structural(x, iSubject);
     case 'ASL'
-        QCCollectFunction = @xASL_qc_CollectQC_ASL;
+		% Check for a session subfield and create when necessary
+		if ~isfield(x.Output.ASL, x.SESSIONS{iSession})
+			x.Output.ASL.(x.SESSIONS{iSession}) = struct;
```

Comment:
Should this be here? This should go to the same location as the creation of the other structs I would say. Why don't we move this to `xASL_qc_Collect_QC_ASL`? That just makes it more "modularly" (because here we just checked the scantype, and scantype-specific code is handled in the individual functions).

Also, we need a warning and workaround when reprocessing existing data, i.e. x.Output.ASL could already exist (but perhaps it is always overwritten?)

Reply:
Good point. I moved it to the function.

I warning is not needed, as we are not overwriting x.Output.ASL, but always only overwritting x.Output.ASL.ASL_X and we want to overwrite by design ;).
See here d7ed87cd

Reply:
So we keep the content of `x.Output.ASL` from previous runs and only add the new ASL_1 ASL_2 ASL_n fields? I thought that you just added a warning saying that the original content is removed :)

Reply:
Yes. `x.Output.ASL.ASL_1`, and `x.Output.ASL.ASL_2` are overwritten. All other tags within `x.Output.ASL` that don't have the format `x.Output.ASL.ASL_N` are removed - that was this warning about.

Is that fine?

Reply:
I can't see the warning anymore quickly, but I would make this more clear perhaps (at least that is what my question above suggests, that it could be more clear :)